jetty-http2/http2-client/src/main/java/org/eclipse/jetty/http2/client/HTTP2ClientConnectionFactory.java

```diff
@@ -58,23 +59,25 @@ public Connection newConnection(EndPoint endPoint, Map<String, Object> context)
 
         Generator generator = new Generator(byteBufferPool, client.getMaxDynamicTableSize(), client.getMaxHeaderBlockFragment());
         FlowControlStrategy flowControl = client.getFlowControlStrategyFactory().newFlowControlStrategy();
-        HTTP2ClientSession session = new HTTP2ClientSession(scheduler, endPoint, generator, listener, flowControl);
+
+        Parser parser = new Parser(byteBufferPool, 4096, 8192);
+        parser.setMaxFrameLength(client.getMaxFrameLength());
+        parser.setMaxSettingsKeys(client.getMaxSettingsKeys());
+
+        HTTP2ClientSession session = new HTTP2ClientSession(scheduler, endPoint, parser, generator, listener, flowControl);
```

Comment:
The parser needs to be built outside because it is configured with the session as the parser listener, and needs to be given to the session for reconfiguration in case a SETTINGS frame is received.
What are we doing for the generator?

Reply:
> What are we doing for the generator?

passing it in after constructing it with configuration pulled from the client.